src/realm/group.cpp

```diff
@@ -1860,6 +1876,34 @@ void Group::advance_transact(ref_type new_top_ref, size_t new_file_size, _impl::
 }
 
 
+void Group::prepare_history_parent(Array& history_root, int history_type,
+                                   int history_schema_version)
+{
+    REALM_ASSERT(m_alloc.get_file_format_version() >= 7);
+    if (m_top.size() < 10) {
+        REALM_ASSERT(m_top.size() <= 7);
+        while (m_top.size() < 7) {
+            m_top.add(0); // Throws
+        }
+        ref_type history_ref = 0; // No history yet
+        m_top.add(RefOrTagged::make_tagged(history_type)); // Throws
+        m_top.add(RefOrTagged::make_ref(history_ref)); // Throws
+        m_top.add(RefOrTagged::make_tagged(history_schema_version)); // Throws
+    }
+    else {
+        int stored_history_type = int(m_top.get_as_ref_or_tagged(7).get_as_int());
```

Comment:
@danielpovlsen Note also that `-Wold-style-cast` warns on c-style casts `(type)value`, not on functional casts.